library/core/src/mem/maybe_uninit.rs

```diff
@@ -252,6 +252,33 @@ use crate::{fmt, intrinsics, ptr, slice};
 ///     std::process::exit(*code); // UB! Accessing uninitialized memory.
 /// }
 /// ```
+///
+/// # Validity
```

Comment:
Moving [this discussion](https://github.com/rust-lang/unsafe-code-guidelines/issues/286#issuecomment-2839284938) here:

> > The `MaybeUninit` docs probably make sense for this. We now do have a [definition of "byte"](https://doc.rust-lang.org/nightly/reference/memory-model.html#bytes) in the reference that this can link to.
> 
> Okay, awesome. And what wording would you recommend? Would it be accurate to say something like the following?
> 
> > The value of a `[MaybeUninit<u8>; N]` may contain pointer provenance, and so `p: P -> [MaybeUninit<u8>; N] -> P` preserves the value of `p`, including provenance

@RalfJung would you like me to add language like this to this PR?

Reply:
Updated to reference the [definition of a byte](https://doc.rust-lang.org/nightly/reference/memory-model.html#bytes).